core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -18,9 +18,19 @@ class SemiCollapsibleNavigationClass {
       window.innerWidth !== 0 &&
       window.innerWidth < CSS_BREAKPOINTS.desktopMinWidth
     ) {
-      this.setCollapsed(true);
+      if (this.isSemiCollapsed === undefined) {
```

Comment:
I find the if statements a bit cluttered here. If it would make sense for you as well I would suggest something like this: 

```
    let isSemiCollapsedUndefined = this.isSemiCollapsed === undefined
    if (
      this.semiCollapsible &&
      window.innerWidth !== 0 &&
      window.innerWidth < CSS_BREAKPOINTS.desktopMinWidth
    ) {
      this.isSemiCollapsed = isSemiCollapsedUndefined ? true : this.getCollapsed();
    } else {
      this.isSemiCollapsed = isSemiCollapsedUndefined ? false : this.getCollapsed();
    }
```

Reply:
Yes, make sense definitely. Thanks.